ca/ca.go

```diff
@@ -432,6 +427,19 @@ func (ca *CA) getTLSConfig(auth *authority.Authority) (*tls.Config, error) {
 	tlsConfig.Certificates = []tls.Certificate{}
 	tlsConfig.GetCertificate = ca.renewer.GetCertificateForCA
 
+	certPool := x509.NewCertPool()
+	for _, crt := range auth.GetRootCertificates() {
+		certPool.AddCert(crt)
+	}
+
+	// adding the intermediate CA to the pool will allow clients that do not
+	// send the intermediate for chain building to connect to the CA successfully.
+	cert, err := auth.GetIntermediateCertificate()
+	if err != nil {
+		return nil, err
+	}
+	certPool.AddCert(cert)
```

Comment:
As an easy fix, without requiring that new method or changes in the CAS services, we can get the cert from the tlsCert from above.

Reply:
We should also include all intermediates, as there might be more than one.

Reply:
Fixed in https://github.com/smallstep/certificates/pull/797/commits/64680bb16dd076a33e3c1ad53b71b507825c6560.